submissions/chris-voitova/friends-app/script.js

```diff
@@ -0,0 +1,206 @@
+           ${email}
+         </a>
+         <span class="user__gender">gender: ${gender}</span>
+         <span class="user__age">age: ${age}</span>
+      </div>
+   </div>`;
+};
+
+async function getUsers() {
+  try {
+    const response = await fetch(URL);
+    const json = await response.json();
+    return json.results;
+  } catch {
+    USERS_CONTAINER.innerHTML += showMessage("oops :( users fetch failed");
+  }
+}
+
+async function renderApp() {
+  const initUsers = await getUsers();
+  const users = initUsers.slice();
+  const sortByAgeParameters = {
+    users: users,
+    sortingTriggersContainer: document.querySelector("#age-sort"),
+    sortingFunction: sortByAge,
+    ascendingTrigger: document.querySelector("#sort-ascending"),
+    descendingTrigger: document.querySelector("#sort-descending"),
+  };
+  const sortByNameParameters = {
+    users: users,
+    sortingTriggersContainer: document.querySelector("#name-sort"),
+    sortingFunction: sortByName,
+    ascendingTrigger: document.querySelector("#sort-az"),
+    descendingTrigger: document.querySelector("#sort-za"),
+  };
+  renderUsers(users);
+  handleSearch(users);
+  handleFilterByGender(users, initUsers);
+  handleSort(sortByAgeParameters);
+  handleSort(sortByNameParameters);
+  handleResetToDefault(initUsers);
+}
+
+const renderUsers = (users) => {
+  let renderedUsers = "";
+  users.forEach(
+    ({
+      picture: { medium: photo },
+      name: { first, last },
+      email,
+      gender,
+      dob: { age },
+    }) => {
+      const name = `${first} ${last}`;
+      renderedUsers += createUserTemplate(photo, name, email, gender, age);
+    }
+  );
+  USERS_CONTAINER.innerHTML = "";
+  USERS_CONTAINER.innerHTML += renderedUsers;
+};
+
+const handleSort = ({
+  users,
+  sortingTriggersContainer,
+  sortingFunction,
+  ascendingTrigger,
+  descendingTrigger,
+}) => {
+  sortingTriggersContainer.addEventListener("click", ({ target }) => {
+    resetSearch();
+    const filteredUsers = checkSelectedState(users);
+
+    if (target === ascendingTrigger) {
+      sortingFunction(filteredUsers);
+      renderUsers(filteredUsers);
+    }
+    if (target === descendingTrigger) {
+      sortingFunction(filteredUsers).reverse();
+      renderUsers(filteredUsers);
+    }
+  });
+};
+
+const checkSelectedState = (users) => {
+  let data;
+  if (selectedGender === "male") {
+    data = filterByGender(users, "male");
+  } else if (selectedGender === "female") {
+    data = filterByGender(users, "female");
+  } else {
+    data = users;
+  }
+  return data;
+};
+const handleFilterByGender = (users, initUsers) => {
+  const maleTrigger = document.querySelector("#gender-male");
+  const femaleTrigger = document.querySelector("#gender-female");
+  const allGendersTrigger = document.querySelector("#gender-all");
+  const genderBlock = document.querySelector("#gender-block");
+
+  genderBlock.addEventListener("click", ({ target }) => {
+    if (target === maleTrigger) {
```

Comment:
You can replace this checks by one query selector, which will get value of checked radio button